### ISSUE TITLE
Promise.mapSeries 异常不会被 catch 捕获

### ISSUE THREAD

**Issue by a573367014** [doodlewind/nativebird#5]
```js
Promise.mapSeries([0, 1, 2], (v) => {
    if (v > 0) throw new Error('111');
    return v;
}).catch(() => {
    // 没触发
    debugger;
});
```

**Comment by a573367014** [comment]
```js
// TODO refactor with Symbol.Iterable
Promise.mapSeries = function mapSeries(arr, fn) {
    if (!Array.isArray(arr)) {
        throw new TypeError(`Promise.mapSeries requires array, but got ${typeof arr}`);
    }
    return new Promise(async (resolve) => {
       // 看起来是这里没加 reject 导致
        const results = [];
            for (let i = 0; i < arr.length; i++) {
                const val = await Promise.resolve(arr[i]);
                results[i] = await fn(val, i, arr.length);
            }
        resolve(results);
    });
};
```

**Comment by doodlewind** [comment]
PR welcome :D

**Comment by doodlewind** [comment]
fixed in `nativebird@1.2.9`
